portfolio/src/main/java/com/google/sps/servlets/ServerCommentsServlet.java

```diff
@@ -0,0 +1,59 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.ReviewerData;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/comments")
+public final class ServerCommentsServlet extends HttpServlet {
+
+  private final List<ReviewerData> reviews = new ArrayList<>();
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get & store the input from the comment section.
+    String reviewerName = getParameter(request, "name", "Anonymous");
+    String reviewerComment = getParameter(request, "comment", "");
+    reviews.add(new ReviewerData(reviewerName, reviewerComment));
+    
+    // Convert the comments to JSON
+    String reviewJson = convertToJson(reviews);
+    System.out.println(reviewerName + ' ' + reviewerComment);
+
+    // Respond with the result.
+    response.setContentType("application/json;");
+    response.getWriter().println(reviewJson);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+    // Convert the server dates to JSON
```

Comment:
server dates?

My thoughts on inline comments:

1) When the code is clear, comments like "Convert to JSON" shouldn't be necessary for the reader to understand what is supposed to happen.

2) When the code is not clear, the code should be rewritten such that the code becomes clear and 1) applies.

3) When comments describe high-level intent, then they may actually add to understanding.